datasync/chaindatafetcher/kafka/kafka.go

```diff
@@ -60,11 +60,23 @@ func NewKafka(conf *KafkaConfig) (*Kafka, error) {
 		return nil, err
 	}
 
-	return &Kafka{
+	kafka := &Kafka{
 		config:   conf,
 		producer: producer,
 		admin:    admin,
-	}, nil
+	}
+
+	err = kafka.CreateTopic(conf.getTopicName(EventBlockGroup))
+	if err != nil {
+		logger.Warn("creating a block group is failed", "err", err)
+	}
+	err = kafka.CreateTopic(conf.getTopicName(EventTraceBroup))
+	if err != nil {
+		logger.Warn("creating a trace group is failed", "err", err)
+	}
```

Comment:
Does this service work when the topics are not created?

Reply:
Yes, there is a broker option to create a topic automatically.

Reply:
Is there any reason you put these lines even though it is not necessary?

Reply:
Ah, it is necessary if a topic should be created with the given parameters. So, the topic wasn't created properly with the chaindatafetcher options. The topic that I used was used previously made. I leave the logs as well.